### PR TITLE
releases: add tidb release notes in 4.0.0-rc.1

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add the TiDB notes to the 4.0.0-rc.1 release notes.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3021
- Other reference link(s):

### CODE REVIEW

File: releases/4.0.0-rc.1.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ TiUP version: 4.0.0-rc.1
 
 ## Important Bug Fixes
 
++ TiDB
+
+    - Fix the issue that the `INSERT ... ON DUPLICATE UPDATE` statement might be incorrectly executed on multiple rows in an explicit transaction because `MemBuffer` is not checked [#16689](https://github.com/pingcap/tidb/pull/16689)
+    - Fix the data inconsistency when locking duplicated keys on multiple rows [#16769](https://github.com/pingcap/tidb/pull/16769)
+    - Fix the panic that occurs when recycling the non-superbatch idle connection between TiDB instances [#16303](https://github.com/pingcap/tidb/pull/16303)
```
Comment (ran-huang) [inline releases/4.0.0-rc.1.md:30]: This note is inconsistent with the Chinese version

File: releases/4.0.0-rc.1.md [hunk 2]
```diff
@@ -89,6 +102,26 @@ TiUP version: 4.0.0-rc.1
...
+- Forbid multiple instances of a TiDB server sharing the same temporary directory [#16026](https://github.com/pingcap/tidb/pull/16026)
+- Fix the panic that occurs during the execution of `explain for connection` when the plan cache is enabled [#16285](https://github.com/pingcap/tidb/pull/16285)
+- Fix the issue that the result of the `tidb_capture_plan_baselines` system variable is incorrectly displayed [#16048](https://github.com/pingcap/tidb/pull/16048)
+- Fix the issue that the `group by` clause in the `prepare` statement is incorrectly parsed [#16377](https://github.com/pingcap/tidb/pull/16377)
+- Fix the panic that might occur during the execution of the `analyze primary key` statement [#16081](https://github.com/pingcap/tidb/pull/16081)
+- Fix the issue that the TiFlash store information in the `cluster_info` system table is wrong [#16024](https://github.com/pingcap/tidb/pull/16024)
+- Fix the panic that might occur during the Index Merge process [#16360](https://github.com/pingcap/tidb/pull/16360)
+- Fix the issue that incorrect result might occur when the Index Merge reader reads the generated columns [#16359](https://github.com/pingcap/tidb/pull/16359)
```
Comment (ran-huang) [inline releases/4.0.0-rc.1.md:116]:
```suggestion
- Fix the issue that an incorrect result might occur when the Index Merge reader reads the generated columns [#16359](https://github.com/pingcap/tidb/pull/16359)
```

File: releases/4.0.0-rc.1.md [hunk 3]
```diff
@@ -89,6 +102,26 @@ TiUP version: 4.0.0-rc.1
...
+- Fix the incorrect display of the default sequence value in the `show create table` statement [#16526](https://github.com/pingcap/tidb/pull/16526)
+- Fix the issue that the `not-null` error is returned because the sequence is used as the default values of the primary key [#16510](https://github.com/pingcap/tidb/pull/16510)
+- Fix the issue that no error is reported for a blocked SQL execution when TiKV continues to return the `StaleCommand` error [#16530](https://github.com/pingcap/tidb/pull/16530)
+- Fix the issue that an error is reported if you only specify `COLLATE` when creating a database; add the missing `COLLATE` part in the result of `SHOW CREATE DATABASE` [#16540](https://github.com/pingcap/tidb/pull/16540)
+- Fix the partition pruning failure when the plan-cache is enabled [#16723](https://github.com/pingcap/tidb/pull/16723)
+- Fix the bug that `PointGet` returns wrong results when handling the overflow [#16755](https://github.com/pingcap/tidb/pull/16755)
+- Fix the issue that wrong result is returned when querying the `slow_query` system table with equal time values [#16806](https://github.com/pingcap/tidb/pull/16806)
```
Comment (ran-huang) [inline releases/4.0.0-rc.1.md:123]:
```suggestion
- Fix the issue that a wrong result is returned when querying the `slow_query` system table with equal time values [#16806](https://github.com/pingcap/tidb/pull/16806)
```

File: releases/4.0.0-rc.1.md (hunk 1 above)
Comment (TomShawn) [reply releases/4.0.0-rc.1.md:30]: @crazycs520 PTAL, the Chinese is `修复回收空闲 gRPC 连接导致 TiDB panic 的问题`. Please confirm whether the English version is OK, thanks!

Comment (crazycs520) [reply releases/4.0.0-rc.1.md:30]: The english version is more accurate,  but the Chinese version also ok.

**Comment by TomShawn** [comment]
@scsldb @crazycs520 PTAL, thanks!

**Pull request merged**